### PR TITLE
Add istioctl analyzer for envoyFilter patch operation

### LINKED ISSUE #37415: Add istioctl analyze warning about fragile EnvoyFilter patch order
(This is used to request new product features, please visit <https://discuss.istio.io> for questions on using Istio)

**Describe the feature request**

An `EnvoyFilter` with an insert position relative to another filter (e.g., `INSERT_BEFORE`) can be
fragile because the default order of filter evaluation is based on the creation time of the filters.
If the insertion point for a filter is not found, the filter will not be added to the chain and nothing will
appear in the error log to indicate that it was skipped.

Documentation describing this subtlety has been added in https://github.com/istio/istio.io/pull/10935.
It would be good to also make `istioctl analyze` produce a warning message for this kind of
fragile configuration.

**Describe alternatives you've considered**

**Affected product area (please put an X in all that apply)**

[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[x] User Experience
[ ] Developer Infrastructure

**Affected features (please put an X in all that apply)**

[ ] Multi Cluster
[ ] Virtual Machine
[ ] Multi Control Plane

**Additional context**

### PR DESCRIPTION
**Pull request by AndreaM12345**
**Please provide a description of this PR:**
Fixes #37415 

Checks an envoy filter patch operation to see if the envoy filter has a priority set or if its using a relative instead of absolute patch operation.  It issues a warning if a relative patch operation (INSERT_BEFORE/INSERT_AFTER) is being used and no priority is being set.

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [x] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: releasenotes/notes/37415.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+apiVersion: release-notes/v2
+kind: feature
+area: istioctl
+releaseNotes:
```
Comment (GregHanson) [inline releasenotes/notes/37415.yaml:4]: Can you format the release note similar to this one?  There is a way to reference the GitHub Issue directly: https://github.com/istio/istio/blob/master/releasenotes/notes/24905.yaml

Comment (ericvn) [reply releasenotes/notes/37415.yaml:4]:
```
issue:
- 37415
```

File: pkg/config/analysis/analyzers/envoyfilter/envoyfilter.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+func (*EnvoyPatchAnalyzer) analyzeEnvoyFilterPatch(r *resource.Instance, c analysis.Context) {
+	ef := r.Message.(*network.EnvoyFilter)
+
+	// if the envoyFilter does not have a priority and it uses the INSERT_BEFORE or INSERT_AFTER
+	// then it can have issues if its using an operation that is relative to another filter.
+	// the default priority for an envoyFilter is 0
+	if ef.Priority == 0 {
+		for index, patch := range ef.ConfigPatches {
+			if patch.Patch.Operation == network.EnvoyFilter_Patch_INSERT_BEFORE || patch.Patch.Operation == network.EnvoyFilter_Patch_INSERT_AFTER {
```
Comment (frankbu) [inline pkg/config/analysis/analyzers/envoyfilter/envoyfilter.go:62]: Why are you only checking for INSERT_BEFORE and INSERT_AFTER? The problem is more general, i.e., for any operation that is relative to another filter. MERGE, DELETE, and REPLACE are the other ones where the order of evaluation is matters.

File: pkg/config/analysis/msg/messages.yaml [hunk 3]
```diff
@@ -555,4 +555,11 @@ messages:
...
     description: "Proxy may prevent tcp named ports and unmatched traffic for ports serving TCP protocol from being forwarded correctly for ExternalName services."
-    template: "Port name for ExternalName service is invalid. Proxy may prevent tcp named ports and unmatched traffic for ports serving TCP protocol from being forwarded correctly"
\ No newline at end of file
+    template: "Port name for ExternalName service is invalid. Proxy may prevent tcp named ports and unmatched traffic for ports serving TCP protocol from being forwarded correctly"
+
+  - name: "EnvoyFilterUsesRelativeOperation"
+    code: IST0151
+    level: Warning
+    description: "This envoy filter does not have a priority and has a relative patch operation set which can cause the envoyFilter not to be applied. Using the INSERT_FIRST option or setting the priority may help in ensuring the envoyFilter is applied correctly"
+    template: "This envoy filter does not have a priority and has a relative patch operation set which can cause the envoyFilter not to be applied. Using the INSERT_FIRST option or setting the priority may help in ensuring the envoyFilter is applied correctly"
```
Comment (frankbu) [inline pkg/config/analysis/msg/messages.yaml:564]: Change:

`Using the INSERT_FIRST option or setting the ...`

to:

`Using an absolute operation (e.g., INSERT_FIRST) or setting the ...`

**Pull request merged**